### PR TITLE
Update UG to reflect new functionalities

### PR DESCRIPTION
**Pull request by kyouscrap**
UG is not updated with the enhanced find features, and is also missing some other key details.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -154,15 +154,15 @@ Symbol     | Parameter    | Constraints
 Symbol     | Parameter    | Constraints
 -----------|--------------|----------------------------------------------------------------------------------------------------------------------------------------------------------
 **`a`**| `DESCRIPTION` |- Must contain at least 1 alphabetic character and has a limit of 40 characters.
-**`s`**| `START`      |- A singular `DATE-TIME` in the form `DD-MM-YYYY-HH-MM`.
-**`e`**| `END`        |- A singular `DATE-TIME` in the form `DD-MM-YYYY-HH-MM`.
+**`s`**| `START`      |- A singular `DATE-TIME` in the form `DD-MM-YYYY-HH-mm`, where `DD` is the day, `MM` is the month, `YYYY` is the year, `HH` is the hour and `mm` is the minute.
+**`e`**| `END`        |- A singular `DATE-TIME` in the form `DD-MM-YYYY-HH-mm`, where `DD` is the day, `MM` is the month, `YYYY` is the year, `HH` is the hour and `mm` is the minute..
```
Comment (choonzies) [inline docs/UserGuide.md:158]: mm is small letters in the form, but capitalised in the description, perhaps we should standardise to all caps?

**Review by niclammm — COMMENTED** [review]
Lgtm

File: docs/UserGuide.md [hunk 2]
```diff
@@ -283,12 +284,14 @@ Valid fields for `find` Command:
 * Ward: Use `w/` to search by ward.
 * Diagnosis: Use `d/` to search by diagnosis.
 * Medication: Use `m/` to search by medication.
+* Patient Note: Use `pn/` to search by patient note.
```
Comment (choonzies) [inline docs/UserGuide.md:287]: Should patient note be singular or plural?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -351,6 +355,8 @@ Format: `makeappt INDEX a/DESCRIPTION s/START e/END`
 * `START` and `END` refers to the date and time the appointment starts and ends respectively.
 * The start **must be before** the end date and time.
 * Appointment added **must not overlap** the duration of existing appointments.
+* Adding an appointment to a patient who already has an existing appointment will cause the existing appointment to be overwritten.
+  * Note that attempting to overwrite an appointment will still require the new appointment to not overlap any existing ones, including the appointment that is about to be overwritten.
```
Comment (choonzies) [inline docs/UserGuide.md:359]: I like this detail!

**Review by choonzies — COMMENTED** [review]
Minor potential typos

**Pull request merged**